src/Controller/GameController.php

```diff
@@ -173,78 +177,87 @@ public function games(): JsonResponse
     }
 
     /**
-     * @Route("/games/{id}/addMarble", name="addMarble")
+     * @Route("/games/{id}/addMarble", name="addMarble", methods={"POST"})
      * Add a marble to the board
      */
 
     public function addMarble(Request $request, string $id, PublisherInterface $publisher): JsonResponse
     {
-        $playerHash = $request->cookies->get($this::COOKIE_KEY);
-
+        $content = $request->toArray();
+        if (!$content || !$content["playerId"] || !$content["position"]) {
+            return new JsonResponse("playerId and marble are required", JsonResponse::HTTP_BAD_REQUEST);
+        }
         $entityManager = $this->getDoctrine()->getManager();
         $game = $entityManager->getRepository(Game::class)->find($id);
 
-        if (!$playerHash || !$game) {
-            return $this->redirectToRoute('newGame');
+        if (!$game) {
+            return new JsonResponse("Game not found", JsonResponse::HTTP_NOT_FOUND);
         }
 
+        $playerId = $content["playerId"];
 
-        if ($game->getStatus() === $this->gameService::GAME_WAITING_OPPONENT ||
-            $game->getCurrentPlayerHash() !== $playerHash
-        ) {
-            return $this->redirectToRoute('game', ["id" => $game->getId()]);
+        // $playerId is a string, we need to convert uuid to the same string.
+        if ($playerId !== $game->getCurrentPlayer()->getId()->toRfc4122()) {
+            return new JsonResponse("This is not your turn", JsonResponse::HTTP_BAD_REQUEST);
```

Comment:
As it is an authentication problem, it would be better to return in code `HTTP_UNAUTHORIZED` or `HTTP_FORBIDDEN`